test/built-ins/Array/prototype/copyWithin/coerced-values-start-change-start.js

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2019 Google. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-array.prototype.copywithin
+description: >
+  SECURITY: start argument is coerced to an integer value
+  and side effects change the length of the array so that
+  the start is out of bounds
+info: |
+  22.1.3.3 Array.prototype.copyWithin (target, start [ , end ] )
+
+  ...
+  8. Let relativeStart be ToInteger(start).
+  ...
+includes: [compareArray.js]
+---*/
+
+
+// make a long integer Array
+function longDenseArray(){
+	var a = [0];
+	for(var i = 0; i < 1024; i++){
+		a[i] = i;
+	}
+	return a;
+}
+
+function shorten(){
+	currArray.length = 20;
+	return 1000;
+}
+
+var array = [];
+array.length = 20;
+
+var currArray = longDenseArray();
+
+assert(
+  compareArray(
+    currArray.copyWithin(0, {valueOf: shorten}), array
+  ),
+  'coercion side-effect makes start out of bounds'
+);
+
+currArray = longDenseArray();
+Object.setPrototypeOf(currArray, longDenseArray());
+
+var array2 = longDenseArray();
+array2.length = 20;
+for(var i = 0; i < 24; i++){
+	array2[i] = Object.getPrototypeOf(currArray)[i+1000];
+}
+
+assert(
+  compareArray(
+    currArray.copyWithin(0, {valueOf: shorten}), array2
+  ),
+  'coercion side-effect makes start out of bounds with prototype'
```

Comment:
With the latest changes, i get 1 failure instead of 2 (here).

What's the state of the current implementation in v8 in latest node; is this expected to pass or fail?

Reply:
Can you link to what you're using to run the tests? I'm running these with: https://github.com/bterlson/test262-harness and getting passes on V8, JSC and Spidermonkey downloaded with JSVU today

Reply:
Sure! I'm running `npm run tests-only` and `npm run test:native` inside https://github.com/ljharb/es6-shim/commit/14866a78ee5b1af3f0ef1aac441c6562311be15e (clone/checkout and `npm install` first)

Reply:
I think this is because deep.equal fails if the prototype of the arrays are different, meanwhile compareArray only compares the elements

Reply:
aha, thanks, i can correct the test then. thanks for looking into it!!!